Library/BillingInfo.cs

```diff
@@ -401,6 +405,13 @@ internal override void WriteXml(XmlTextWriter xmlWriter)
                     xmlWriter.WriteElementString("external_hpp_type", ExternalHppType.Value.ToString().EnumNameToTransportCase());
 
                 }
+
+                if (!GatewayCode.IsNullOrEmpty())
+                {
+                    xmlWriter.WriteElementString("gateway_code", GatewayCode);
+                    xmlWriter.WriteElementString("gateway_token", GatewayToken);
+                    xmlWriter.WriteElementString("card_type", CardType.ToString().ToLower());

```

Comment:
Will this produce the correct string for every value? You may need to use `EnumNameToTransportCase` here.

Reply:
Fixed